vault/expiration.go

```diff
@@ -120,13 +129,37 @@ func (c *Core) stopExpiration() error {
 	return nil
 }
 
+// restoreLock takes out a lock only when in restore mode and returns if in
+// restore mode
+func (m *ExpirationManager) restoreLock() {
+	if m.inRestoreMode() {
+		m.restoreMutex.Lock()
+	}
+}
+
+// restoreUnlock unlocks only when in restore mode
+func (m *ExpirationManager) restoreUnlock() {
```

Comment:
I'm concerned that -- maybe not now, but with simple logic changes down the line -- a process doing a sync restore of a lease takes out a lock, restore mode shuts down in the interim, and then it never unlocks. I would suggest that restoreLock() goes away and it's simply the responsibility of the calling process to Unlock the mutex when done.

Reply:
Although this does suggest that the comment on restoreLock needs to actually be true :-)

I would suggest that if it's in restore mode, restoreLock locks the mutex and returns the pointer to the lock. If a calling process gets a nil lock back, it knows it doesn't need to unlock later.

Reply:
This was just to prevent a bunch of copy/paste into every method in the expiration manager.  You are right and I will just put the check in all the calls.

Reply:
I think this is dangerous. If the restore mode changes between locking and unlocking with these helper functions we could deadlock, or oppositely we could panic by unlocking a non-locked mutex.